javascript/javascript1/week4/homework6/scripts.js

```diff
@@ -0,0 +1,133 @@
+        const match = command.match(regex) // this method will return an array with matching values to the regex 
+        return match ? Number(match[0]) : null; // if true convert the regex to a number and return it
+    }
+    const number = getInterval(command)
+    const interval = number * 60 * 1000;
+    function logDone(){
+        console.log(`Timer Done`)
+    }
+    setTimeout(logDone, interval)
+    console.log(`A timer for ${number} minute(s) has been set`)
+}
+
+
+
+//main function to get answers from the "voice assistant"
+function getReply(command) {
+    // here we deal with the greeting commands
+    if (command.startsWith('Hello my name is')){
+        const person = getName(command)
+        if (!loggedName.includes(person)) {
+            loggedName.push(person)
+            return console.log(`Nice to meet you ${person}`)
+        } else if (loggedName.includes(person)) {
+            return console.log(`Hey ${person}! we met already.`)
+        }
+    }
+    // here we deal with reminding the user their name
+    else if (command.startsWith('What is my name')) {
+        if (loggedName.length === 0) {
+            return console.log(`I can't find a name. \n Please introduce yourself in this way: "Hello my name is" followed by your name`)
+        } else if (loggedName.length > 0) {
+            const recentName = loggedName[loggedName.length -1]
+            return console.log(`Your name is ${recentName}! What can I help you with, ${recentName}?`)
+        }
+    }
+    //dealing with the todo part
+    else if (command.includes('my todo')) {
+        let newTask = getTask(command);
+        //adding a task
+        if(command.startsWith('Add') || command.startsWith('add')){
+           todos.push(newTask); 
+           return console.log(`${newTask} has been added to your todos`)
+        
+        } 
+        //removing a task
+        else if (command.startsWith('Remove') || command.startsWith('remove')){
+            const taskIndex = todos.indexOf(newTask);
+            todos.splice(taskIndex, 1)
+            return console.log(`${newTask} has been removed from your todos`)
+        }
+        // showing all task 
+        else if (command.startsWith('What is on my todo?')){
+            if (todos.length === 0) {
+                console.log(`Your list is empty, try adding some tasks first`)
+            } else {
+                console.log(`Your todo list is comprised of: \n${todos.join('\n')}`)
+            }
+        }
+    }
+    // dealing with the date 
+    else if (command.startsWith('What day is it today?') || command.startsWith('What day is it') ) {
+       return getDate()
+    }
+    // dealing with math operations
+    else if ( command.startsWith('What is') && (/[+\-*/]/.test(command))){ //apparently .includes does not work with regex, so instead you se .test
+        const operation = command.slice(7)
+        return eval(operation)
```

Comment:
While `eval()` is very useful for cases like this where we want to evaluate an expression, we need to be very careful about using it as developers.

Many sites will disable using it entirely these days, but consider what happens if the user types
```
what is alert(`hello ${loggedName[-1]}`)
```

That's right - this user input will cause the browser (if running in the browser) to pop up an alert dialog, and it's even got access to the value of any variables! Consider that this could be used to send information about the user to a malicious site, and you'll start to realise why `eval()` is generally considered best left alone!

That said, for this assignment, it's a great way to get started.

An alternative could be to parse some numbers and an operator (+, -, *, /) form the command string, and then perform a different operation based on the operator.

Reply:
Woooooaaaaaaw 
That is awesome knowledge!!
Thank you so much. I will try the parsin method